src/template.rs

```diff
@@ -143,11 +175,27 @@ impl ObjectTemplate {
         }
     }
 
+    /// Creates a new object instance based off of this template with an internal rust object accessible from callbacks.
+    pub fn new_instance_with_internal(&self, context: &context::Context, internal: T) -> value::Object {
```

Comment:
The safety of this depends on if the external check in the callback works properly. I'll add more to that comment.